src/main/scala/models/Customers.scala

```diff
@@ -0,0 +1,9 @@
+package models
+
+import models.Domain.{Customer, CustomerId}
+
+trait Customers[F[_]] {
```

Comment:
**never** do interfaces (traits) for domain service concepts:
- `Customers` is pure business logic
- there is no gain in interfacing business logic because u **never** have multiple implementations of business logic
- that is the part that u want to test, so what is the reason to make it `trait`
- maybe (?) to facilitate mocking (can be not realistic at all that why u interface data provider levels and NOT business logic)

Reply:
that class => completely remove replace with CustomerService[F[_]](repository: CustomerRepository)